client/lib/abtest/index.js

```diff
@@ -64,7 +63,7 @@ export const saveABTestVariation = ( name, variation ) =>
 
 export const getAllTests = () => keys( activeTests ).map( ABTest );
 
-const isUserSignedIn = () => user && user.get() !== false;
+const isUserSignedIn = () => user() && user().get() !== false;
```

Comment:
`user()` is always guaranteed to return a valid `User` object. No need to check it. It's only the `user().get()` return value, i.e., `this.data` of the `User` singleton instance, that can be `false` for logged-out session.

Reply:
That's sadly not true. I had assumed that at first but it broke the server side. Unfortunately, the server build aliases `lib/user` to a no-op, which means that `user()` ends up returning `undefined`, hence the need for the check.

Reply:
🤦‍♂️ It's very likely that the server alias for `lib/user` is obsolete. `user.get()` used to trigger `user.fetch()`, which would be unwanted on server, but that doesn't happen any more.

And if we remove the static module-level calls to `user()` in this PR, that removes another reason why server-side `lib/user` could be harmful.

Reply:
By the way, it seems you do the `user()` check only in `lib/abtest` and nowhere else. Is `lib/abtest` really the only module affected by the server-side alias?

Reply:
There are other places, I'm afraid 😕 I restored the check to all the places where it previously existed, in this commit, because I didn't want to untangle which ones were really needed and which ones weren't: https://github.com/Automattic/wp-calypso/pull/43069/commits/ccf62e9a04236196d3a93fd3adb63b990fe784c8

it includes signup and initial state as well.

What do you think the right course of action is here? Remove the alias and the checks as part of this PR as well?

Reply:
> What do you think the right course of action is here?

Remove the alias and the check, but it totally deserves a standalone PR. This one is already snowballing into areas you probably didn't plan to address in the beginning.

Reply:
No problem! I'll prepare another PR with those changes, then 🙂 Thanks!